app/models/dscc_number.rb

```diff
@@ -0,0 +1,29 @@
+class DsccNumber < ActiveRecord::Base
+
+  def self.current_monthly_number
+    where(year_and_month: current_year_and_month).order('number desc').limit(1).pluck(:number).first || 0
+  end
+
+  def self.next_monthly_number
+    current_monthly_number + 1
+  end
+
+  def self.current_year_and_month
+    Time.current.to_date.beginning_of_month
```

Comment:
The DSCC number is keyed off when the DefenceRequest is raised, not when the DsccNumber is generated.
If the DefenceRequest is raised just before midnight on the last day of the month then its DsccNumber must be part of the same month. 
Hence storing the year_and_date separately to DsccNumber#created_at